chips/stm32f4xx/src/adc.rs

```diff
@@ -264,7 +264,7 @@ const ADC_COMMON_BASE: StaticRef<AdcCommonRegisters> =
 
 #[allow(dead_code)]
 #[repr(u32)]
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, PartialEq)]
```

Comment:
Why do we need the `PartialEq`?

Reply:
Several apps can request values from the same channel, there is an if condition to reply with the same sample at the same time to them.

```rust
impl<'a, A: hil::adc::Adc> hil::adc::Client for MuxAdc<'a, A> {
    fn sample_ready(&self, sample: u16) {
        for node in self.devices.iter() {
            self.inflight.map(|inflight| {
                if node.channel == inflight.channel {
                    node.operation.map(|operation| match operation {
                        Operation::OneSample => {
                            node.operation.clear();
                            node.client.map(|client| client.sample_ready(sample));
                        }
                    });
                }
            });
        }
        self.inflight.clear();
        self.do_next_op();
    }
}
```